### PR TITLE
Prasad_C_Assignment_8

### PR DESCRIPTION
**Pull request by PrasadCOct2020**
Hi Harsh,

Please review my PR.

Output:- 
image

Regards 
Prasad Chitale

### CODE REVIEW

**Comment by PrasadCOct2020** [comment]
Code has below incorrect SOP statement at line no 72
	System.out.println("Step 5: Printing browser window handle names");

File: src/prasad/MissingLink.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		driver.get(
+				"C:\\Users\\Prasad Chitale\\Documents\\java_selenium\\SeleniumTechnoOct2020\\resources\\forms\\MissingLink.html");
+		driver.manage().window().maximize();
+		driver.manage().timeouts().implicitlyWait(2, TimeUnit.SECONDS);
+
+		return driver;
+	}
+
+	void findMissing(WebDriver driver) {
+		ArrayList<WebElement> hyperLinks = new ArrayList<WebElement>();
```
Comment (harshvegada) [inline src/prasad/MissingLink.java:36]: Hi @PrasadCOct2020 
you should use findElements(//a); using this you can get all the elements in list, may be your code will not for other elements.

File: src/prasad/MissingLink.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		ArrayList<WebElement> missingLinks = new ArrayList<WebElement>();
+
+		for (WebElement e : hyperLinks) {
+			String hrefVal = e.getAttribute("href");
+			if (hrefVal == null || hrefVal.length() == 0) {
+				missingLinks.add(e);
+			}
+		}
+		
+		System.out.println("Step 2: Missing link text:-");
```
Comment (harshvegada) [inline src/prasad/MissingLink.java:60]: with single loop you can check weather condition match expectations or not

**Review by harshvegada — APPROVED** [review]
Please work on comments & improve for next assignments .
good to merge

**Pull request merged**